### ISSUE TITLE
Error handling

### ISSUE THREAD

**Issue by 2xmax** [konstantinvlasenko/PowerSlim#68]
Summary

1. Bug: It prints only first error in log.
2. To be discussed: $error result can be highlighted.
3. To be discussed: It stops the execution if an exception occurred.
4. Solution: Let's simplify Exec-Script a little bit.

Discussion:
###### 1. Bug: It prints only first error in log. 
Just let us fix them.

###### 2. To be discussed: $error result can be highlighted

<p>Very often, presence of $error is rather unusual situation. Particularly, with the current implementation I overlooked some quite critical errors. In such a way I recommend to mark instruction and entire test with yellow color in this case.</p>
<p>Another problem deals with legacy suites that may become yellow after the fix. Given that, I suggest to use a flag. Do you have other ideas?</p>

###### 3. To be discussed: It stops the execution if any exception occurred
<p>I did a few small experiments by observing behavior of other Slim implementations in case of exception. As a result, I found that original FitNesse client (Java) and WafferSlim (Python) does not stop the entire execution in case of exception, whereas PowerSlim go another way for some reason. </p>
<p>Accordingly, I would like to recommend to use this approach:
If there is an exception, let acknowledge test runner about the occurred exception (make it yellow), but please don't stop the entire test execution.</p>

###### 4. Solution: Let's simplify Exec-Script a little bit.
I love PowerSlim for its simplicity and I am just amazed at how such a small script can do so much useful work. However, let us take a look at [Exec-Script](https://github.com/konstantinvlasenko/PowerSlim/blob/master/slim.ps1#L239-L320): it has about 100 lines of code, nested conditions and whatnot. So, I would like to show you that it can be smaller. IMHO This as easy as pie implementation contains everything you need (incl. fix of 1-3) and this one is really proved in combat. 10 LOC, that is it!:

```powershell```
function Exec-Script( $Script ) {
   try { iex $Script } catch { "__EXCEPTION__: Unhandled Exception $($_.Exception.ToString()) " }
   if($Error[0] -ne $null) { Print-Error }
}

function Print-Error {
    $Error | % {
        "__EXCEPTION__: Error " + ($_ | Out-String)
        if($_.Exception) { $_.Exception.ToString() }
    } | Out-String
}
```

So, if everyone agree with that, I will prepare a pull request for that.

**Comment by konstantinvlasenko** [comment]
I am always for the simplest possible solution and implementation!
Please, check the discussion for pull request https://github.com/konstantinvlasenko/PowerSlim/pull/52

**Comment by 2xmax** [comment]
Hello, Konstantin. 

I've already checked the pull request. Let I add some comments about that. In addition to exception handling (nevertheless, it was a good improvement), this pull request contains the implementation ABORT_SLIM_TEST specification that seems a good initial attempt for that, but it could be a little bit better. To explain that, let us review the [original specification of the protocol](http://www.fitnesse.org/FitNesse.FullReferenceGuide.UserGuide.WritingAcceptanceTests.SliM.SlimProtocol):

>##### Aborting a Test
> If a fixture throws an exception with a class name that contains **"StopTest"**, then Slim should stop executing the instructions in the current batch, and return immediately. The response for this type of exception should be "__EXCEPTION__:ABORT_SLIM_TEST:" which may have an optional suffix of: "message:<<reason>>".

<p>Digging deeper, let us investigate why both Java and Python implementation will not stop the execution if a regular exception occurs. Having read a little closer we can come to a conclusion, that the executor may throw a special exception (like "throw 'StopTest'" in PowerShell) to abort the suite and that's ok for the special exception, but it must not abort the entire exception for **every** exception.</p>

To sum up, I would propose an extra one-line condition like if the exception contains StopTest, then the executor have to abort test. By the way, I haven't used this Slim feature so far and the implementation I presented have being used for about a year successfully.

**Comment by konstantinvlasenko** [comment]
Looks good for me. Please, create pull request

**Comment by maksim-dell** [comment]
@ALuckyGuy, could you tell me is there any specific assumptions under 'System.Management.Automation.RuntimeException'? Can it just simplify to that (not final code, just prototype)?:

```powershell
function Exec-Script( $Script ) {
   if ($script:SLIM_ABORT_SUITE) { return "__EXCEPTION__:ABORT_SLIM_SUITE:message:<<ABORT_TEST_INDICATED:Test not run>>" }
   if ($script:SLIM_ABORT_TEST)  { return "__EXCEPTION__:ABORT_SLIM_TEST:message:<<ABORT_TEST_INDICATED:Test not run>>" }
   try { 
        iex $Script 
   } catch { 
        $errorType = $Error[0].FullyQualifiedErrorId
        if($errorType -match "'^Stop(Test|Suite):?(.*)?'"){
            # test or suite can be aborted by throw "StopTest"            
            $script:SLIM_ABORT_TEST = $true
            if ($matches[1] -eq 'Suite') {
              $script:SLIM_ABORT_SUITE = $true
            }
            "__EXCEPTION__:ABORT_SLIM_TEST: Additional Info[ $($_.Exception.ToString()) ]"
        } else {
            "__EXCEPTION__: Unhandled Exception $($_.Exception.ToString()) "
        }
   }
   if($Error[0] -ne $null) { Print-Error }
}
```
